test_unstructured/documents/test_ontology_to_unstructured_parsing.py

```diff
@@ -0,0 +1,146 @@
+                ]
+            ),
+        ]
+    )
+
+    unstructured_elements = ontology_to_unstructured_elements(ontology)
+    body, page1, p1, page2, c1, p2, c2, p3 = unstructured_elements
+    assert body.metadata.category_depth == 0
+
+    assert page1.metadata.category_depth == 1
+    assert page2.metadata.category_depth == 1
+
+    assert p1.metadata.category_depth == 2
+
+    assert c2.metadata.category_depth == 2
+    assert c1.metadata.category_depth == 2
+
+    assert p2.metadata.category_depth == 3
+    assert p3.metadata.category_depth == 3
+
+
+def test_chunking_is_applied_on_elements():
+    ontology = Document(
+        children=[
+            Page(children=[Paragraph(text="Paragraph1")]),
+            Page(
+                children=[
+                    Column(children=[Paragraph(text="Paragraph2")]),
+                    Column(children=[Paragraph(text="Paragraph3")]),
+                ]
+            ),
+        ]
+    )
+
+    unstructured_elements = ontology_to_unstructured_elements(ontology)
+
+    chunked_basic = chunk_elements(unstructured_elements)
+    assert str(chunked_basic[0]) == "Paragraph1\n\nParagraph2\n\nParagraph3"
+    chunked_by_title = chunk_by_title(unstructured_elements)
+    assert str(chunked_by_title[0]) == "Paragraph1\n\nParagraph2\n\nParagraph3"
+
+
+def test_embeddings_are_applied_on_elements(mocker):
+    ontology = Document(
+        children=[
+            Page(children=[Paragraph(text="Paragraph1")]),
+            Page(
+                children=[
+                    Column(children=[Paragraph(text="Paragraph2")]),
+                    Column(children=[Paragraph(text="Paragraph3")]),
+                ]
+            ),
+        ]
+    )
+
+    unstructured_elements = ontology_to_unstructured_elements(ontology)
+    # Mocked client with the desired behavior for embed_documents
+    mock_client = mocker.MagicMock()
+    mock_client.embed_documents.return_value = [1, 2, 3, 4, 5, 6, 7, 8]
+
+    # Mock get_client to return our mock_client
+    mocker.patch.object(OpenAIEmbeddingConfig, "get_client", return_value=mock_client)
+
+    encoder = OpenAIEmbeddingEncoder(config=OpenAIEmbeddingConfig(api_key="api_key"))
+    elements = encoder.embed_documents(
+        elements=unstructured_elements,
+    )
+
+    assert len(elements) == 8
+
+    document, page1, p1, page2, c1, p2, c2, p3 = elements
+
+    assert p1.embeddings == 3
+    assert p2.embeddings == 6
+    assert p3.embeddings == 8
+
+
+@pytest.mark.parametrize(
+    ("html_file_path", "json_file_path"),
+    [
+        ("html_files/example.html", "structured_jsons/example.json"),
+    ],
+)
+def test_ingest(html_file_path, json_file_path):
+    html_code = Path(html_file_path).read_text()
+    expected_json_elements = elements_from_json(str(Path(json_file_path)))
+
+    ontology = parse_html_to_ontology(html_code)
+    unstructured_elements = ontology_to_unstructured_elements(ontology)
```

Comment:
for consumers that just want unstructured elements, these two functions calls combined `parse_html_to_ontology()` and `ontology_to_unstructured_elements()`, i.e. html -> unstructured elements should accessible to consumers using [paritition_html()](https://github.com/Unstructured-IO/unstructured/blob/a11ad22609ca016f8080646c20e6da9ad5e7dc08/unstructured/partition/html/partition.py#L23) (or plain old [partition()](https://github.com/Unstructured-IO/unstructured/blob/a11ad22609ca016f8080646c20e6da9ad5e7dc08/unstructured/partition/auto.py#L25) which may call `partition_html()`)  with an additional flag that indicates whether to parse with the prior or new parsing method.